app/forms/openid_connect_logout_form.rb

```diff
@@ -29,7 +29,9 @@ class OpenidConnectLogoutForm
             },
             if: :reject_id_token_hint?
   validates :post_logout_redirect_uri, presence: true
-  validates :state, presence: true, length: { minimum: RANDOM_VALUE_MINIMUM_LENGTH }
+  validates :state,
+            length: { minimum: RANDOM_VALUE_MINIMUM_LENGTH },
+            if: -> { state.present? }
```

Comment:
```suggestion
            if: -> { !state.nil? }
```

There's kind of a narrow gap that I'm not sure whether we want to close, but someone could pass `state=` which would validate successfully.  My sense is to fully validate the parameter if it is supplied in any form, so I've suggested switching the check from `present?` to `!nil?` which will cause us to validate in the case where the value of `state` is `""`

Reply:
ooh good catch, thanks!